modules/order/src/Form/UseMultiProfileConfirmForm.php

```diff
@@ -0,0 +1,390 @@
+          '@modules' => implode('<br>', $incompatible_modules)
+        ]
+      ));
+
+      return;
+    }
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\commerce_order\Entity\OrderTypeInterface $order_type */
+    $order_type = $this->entity;
+
+    // Create the billing and shipping profile types.
+    $this->createBillingShippingProfiles();
+
+    // Migrate any fields added to the customer profile type to the billing and
+    // shipping profile types.
+    $this->migrateProfileFields();
+
+    // Batch process to migrate the existing order profiles from the customer
+    // profile type to use the billing/shipping profile types.
+    $this->migrateExistingProfiles();
+
+    // Set the useSingleProfile field to FALSE now that we've processed
+    // everything.
+    $order_type->setUseSingleProfile(FALSE);
+    $order_type->save();
+
+    $form_state->setRedirectUrl($order_type->toUrl('collection'));
+  }
+
+  /**
+   * Create the billing and shipping profile types, if not already created.
+   */
+  protected function createBillingShippingProfiles() {
+    $profile_type_storage = $this->entityTypeManager->getStorage('profile_type');
+    /** @var \Drupal\profile\Entity\ProfileTypeInterface $billing_profile_type */
+    $billing_profile_type = $profile_type_storage->load('customer_billing');
+
+    // Import YAML config.
+    $config_path = drupal_get_path('module',
+        'commerce_order') . '/config/shipping_billing_profiles';
+    $source = new FileStorage($config_path);
+    $config_storage = $this->configStorage;
+
+    if (!$billing_profile_type) {
+      $billing_configs = [
+        'core.entity_form_display.profile.customer_billing.default',
+        'core.entity_view_display.profile.customer_billing.default',
+        'field.field.profile.customer_billing.address',
+        'profile.type.customer_billing',
+      ];
+      foreach ($billing_configs as $config_name) {
+        $config_storage->write($config_name, $source->read($config_name));
+      }
+    }
+
+    /** @var \Drupal\profile\Entity\ProfileTypeInterface $shipping_profile_type */
+    $shipping_profile_type = $profile_type_storage->load('customer_shipping');
+
+    if (!$shipping_profile_type) {
+      $shipping_configs = [
+        'core.entity_form_display.profile.customer_shipping.default',
+        'core.entity_view_display.profile.customer_shipping.default',
+        'field.field.profile.customer_shipping.address',
+        'profile.type.customer_shipping',
+      ];
+      foreach ($shipping_configs as $config_name) {
+        $config_storage->write($config_name, $source->read($config_name));
+      }
+    }
+
+    if (!$billing_profile_type || !$shipping_profile_type) {
+      $this->messenger()->addMessage($this->t('Billing and shipping profile
+        types have been successfully created.'
+      ));
+    }
+  }
```

Comment:
**OPTIONAL CHANGE REQUEST**

Why are you repeating yourself so much here?

```
protected function createProfileType($profile_type_id) {
  $profile_type = $this->entityTypeManager->getStorage('profile_type')->load($profile_type_id);
  if ($profile_type) {
    return;
  }

  // Import YAML config.
  $config_path = drupal_get_path('module', 'commerce_order') . '/config/shipping_billing_profiles';
  $source = new FileStorage($config_path);

  $configs = [
    "core.entity_form_display.profile.$profile_type_id.default",
    "core.entity_view_display.profile.$profile_type_id.default",
    "field.field.profile.$profile_type_id.address",
    "profile.type.$profile_type_id".
  ];
  foreach ($configs as $config_name) {
    $this->configStorage->write($config_name, $source->read($config_name));
  }
}
```

Reply:
Yeah, that was an air-headed function.